src/libraries/System.Private.CoreLib/src/System/Object.cs

```diff
@@ -47,7 +47,7 @@ public Object()
         /// <returns><c>true</c> if the specified object is equal to the current object; otherwise, <c>false</c>.</returns>
         public virtual bool Equals(object? obj)
         {
-            return RuntimeHelpers.Equals(this, obj);
+            return ReferenceEquals(this, obj);
```

Comment:
can be just `return this == obj;` I guess (shorter and saves time for inliner)

Also I assume you can delete the `RuntimeHelpers.Eqauls` FCall now

Reply:
Actually you can rewrite it to C# and delete the FCall, thus, ILLink will be able to trim it out if it's unused.